src/validate-file.ts

```diff
@@ -0,0 +1,62 @@
+import StreamZip from 'node-stream-zip'
+import type { ErrorObject } from 'ajv'
+import util from 'util'
+
+import { getAjvInstance } from './utils'
+import schema from './file-format.schema.json'
+
+const fromFile = async (filepath: string): Promise<any> => {
+  const archive = new StreamZip({
+    file: filepath,
+    storeEntries: true,
+  })
+
+  const contents = await new Promise((resolve): void => {
+    archive.on('ready', (): void => {
+      const document = JSON.parse(
+        archive.entryDataSync('document.json').toString(),
+      )
+      const pages = document.pages.map((page: { _ref: string }): void =>
+        JSON.parse(archive.entryDataSync(`${page._ref}.json`).toString()),
+      )
+      resolve({
+        document: {
+          ...document,
+          pages,
+        },
+        meta: JSON.parse(archive.entryDataSync('meta.json').toString()),
+        user: JSON.parse(archive.entryDataSync('user.json').toString()),
+      })
+    })
+  })
+
+  archive.close()
+
+  return contents
+}
```

Comment:
Would `fromFile` be better placed somewhere else for others to use?

Reply:
Definitely - but not sure it belongs here with the schemas 🤔

Reply:
Maybe add a `TODO: …` for revisiting later?

Reply:
Yeap.